pymatgen/core/structure.py

```diff
@@ -764,6 +764,28 @@ def get_neighbors(self, site, r, include_index=False):
                                       include_index=include_index)
         return [d for d in nn if site != d[0]]
 
+    def get_nearest_site(self, coords, site, r = None):
```

Comment:
They are not the same. pbc_shortest_vector assumes the same lattice. get_min_distances uses the individual lattices to find the closest image, making comparisons with cartesian coordinates. I went ahead and added PolarizationLattice(Structure) to pymatgen.analysis.ferroelectricity.polarization and removed get_nearest_site from Structure.